### ISSUE TITLE
Resources referencing arrays of sub-resource will break when the child resource script file is moved

### ISSUE THREAD

**Issue by wilbefast** [godotengine/godot#93801]
### Tested versions

4.2.2

### System information

Godot v4.2.2.stable - Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1050 Ti (NVIDIA; 32.0.15.5599) - Intel(R) Core(TM) i7-8700 CPU @ 3.20GHz (12 Threads)

### Issue description

Resource .tres files which reference an array of sub-resources are not automatically updated when the script file for the sub resource file is moved. For example:

```
[resource]
script = ExtResource("1_lnayw")
cards = Array[Resource("res://ResourceScripts/StaticData/CardData.gd")]([])
```

Needs to manually be updated to

```
[resource]
script = ExtResource("1_lnayw")
cards = Array[Resource("res://ResourceScripts/StaticData/Cards/CardData.gd")]([])
```

If the CardData script is moved into the Cards folder. This can be very hard to understand and to fix, and moving files around as a project gets bigger is vital for maintaining a coherent project structure.

### Steps to reproduce

1. Load the attached project
2. Run it
3. Note the lack of error
4. Drag the script ChildResourceScript.gd into the folder MoveChildResourceScriptHereToBreakProject
5. Run the project again
6. Note that it crashes

[MigrationTest.zip](https://github.com/user-attachments/files/16048959/MigrationTest.zip)

### Minimal reproduction project (MRP)

[MigrationTest.zip](https://github.com/user-attachments/files/16048959/MigrationTest.zip)

**Comment by wilbefast** [comment]
@AThousandShips I'm not very familiar with the tags, but looking at the others marked "enhancement" it feels like this is maybe a little more serious: I've really been leaning into the use of Resources for both static and dynamic game data, so have been spending a lot of time setting up all of my data in the editor: repeatedly needing to delete it all and start again from scratch, because it has mysteriously become corrupted, has been very costly. 

The above is no longer a huge issue for me personally, because I've found a work-around (modifying the reference by hand in text), but this took a lot of head-scratching: the error messages and dependency issues that Godot reports don't handle references within Resource files, so you're kind of on your own to figure this one out if it occurs. I'd imagine some folks will look at this problem and say "if I can't move files without breaking my project, then this is not a production-ready tool".

**Comment by AThousandShips** [comment]
The reason is that the handling of script file moving is still complicated and I initially classified it as a missing feature and not a bug, the relevant teams might relabel it if they think it's a bug rather than a missing feature

**Comment by wilbefast** [comment]
Fair enough - I _have_ noticed that moving files is a risky business at the moment: I believe there are a couple of other related issues, but I haven't identified the exact repro steps for these yet.
